src/server/util/TermboxTextOutputArea.cpp

```diff
@@ -0,0 +1,39 @@
+#include "TermboxTextOutputArea.hpp"
+
+void TermboxTextOutputArea::setLines() {
+    uint16_t fg = 0;
+    for (uint32_t y_offset=0; y_offset < m_max_height && 
+            (y_offset < m_line_counter); y_offset++) {
+        uint32_t true_y = m_origin_y + y_offset;
+        if (m_line_counter % m_max_height == y_offset) {
+            fg = TB_BLUE;
+        }
+        setStrAtCoord(m_lines[y_offset], m_origin_x, true_y, fg); 
+    }
+}
+
+void TermboxTextOutputArea::addLine(std::string line) {
+   m_lines[m_line_counter % m_max_height] = line;
+   m_line_counter++;
+}
+
+void TermboxTextOutputArea::setOrigin(uint32_t x, uint32_t y) {
+    m_origin_x = x;
+    m_origin_y = y;
+}
+
+void TermboxTextOutputArea::setMaxHeight(uint32_t max_height) {
+    if (max_height < MAX_LINES) {   
+        m_max_height = max_height;
+    } else {
+        std::cerr << "You must set a max height less than MAX_LINES" << std::endl;
+    }
+}
+
+uint32_t TermboxTextOutputArea::setStrAtCoord(std::string str, uint32_t x, uint32_t y, uint16_t fg) {
```

Comment:
maybe create termbox util since this function is shared with other termbox component

Reply:
decided not to, not enough duplication to be worth the effort